### PR TITLE
Enable users to clip track play times

### PR DESCRIPTION
**Pull request by croasus**
Clip settings appear with other track settings (move to top, etc.) once
a track is selected. Only one track can be clipped at a time.

### CODE REVIEW

File: views/queue.jade [hunk 1]
```diff
@@ -376,6 +572,38 @@ script.
...
+  $('#confirm-clip').on('click', function(e) {
+    submitClipSettings();
+    return false;
+  });
+
+  // Reset track clip
+  $('#reset-clip').on('click', function(e) {
+    var indices = selectedTracks() || {};
+    var pl = selectedPlaylist() || {};
+    if((indices.length != {}) && (pl.length != {})) {
```
Comment (aromatt) [inline views/queue.jade]: do you mean `if ((indices.length > 0) && (pl !== {})) {` ?

File: views/queue.jade [hunk 2]
```diff
@@ -376,6 +572,38 @@ script.
...
+  
+  // Confirm track clip
+  $('#confirm-clip').on('click', function(e) {
+    submitClipSettings();
+    return false;
+  });
+
+  // Reset track clip
+  $('#reset-clip').on('click', function(e) {
+    var indices = selectedTracks() || {};
```
Comment (aromatt) [inline views/queue.jade]: I think this should default to `[]` not `{}`, which is what `selectedTracks()` already returns if no tracks are selected.  Right?

**Comment by aromatt** [comment]
We talked about disabling these controls if more than one track is selected--did you end up doing that?  The exception I could see would be the 'reset' button.  I think when multiple tracks are selected, I don't think you should be able to enter clip values, or click the submit button, but you should be able to reset.  Does that sound right?

File: views/queue.jade (hunk 1 above)
Comment (croasus) [inline views/queue.jade]: Actually I meant for both to be ` != {}`. The reason I put that was 1) I thought I had gotten this sequence/check from somewhere else in the code that you had written, but I am not finding that now so it must have been from early on in my dev here, and 2) I need to double check, but I remember thinking seeing that `selectedTracks` would just come back with one single value if one box is checked, making `.length` not work. Figured it was some weird javascript thing. But I'll double check - I may have been just printing out indices[0] or something.

**Comment by croasus** [comment]
Yes, it does disable (in `updateQueueButtons()`). Yeah, allowing resetting sounds right - currently it disables that also but that's easy to change.

File: views/queue.jade (hunk 2 above)
Comment (croasus) [inline views/queue.jade]: Works for me.

File: views/queue.jade (hunk 1 above)
Comment (aromatt) [inline views/queue.jade]: Also, I just remembered that checking if an object is empty using `!= {}` or even `!== {}` may not work (I probably have these in my code, and those are bugs). You'll have to do `Object.keys(foo).length > 0`
https://glot.io/snippets/eezaky6a3d

**Pull request closed without merging**